src/main/java/hudson/matrix/MatrixBuild.java

```diff
@@ -399,7 +399,15 @@ private void listUpAggregators(Collection<?> values) {
 
         public void post2(BuildListener listener) throws Exception {
             for (MatrixAggregator a : aggregators)
-                a.endBuild();
+                // all aggregations will be performed even any of them failed.
+                try {
+                    if (!a.endBuild()) {
+                        listener.error(String.format("Aggregation failed for %s", a.toString()));
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace(listener.error(String.format("Aggregation failed for %s", a.toString())));
```

Comment:
I think `MatrixAggregator#endBuild()` doesn't throw `AbortException`.
Do you mean `MatrixAggregator#endBuild()` should throw `AbortException` instead of returning false?

Or if you mean we should handle `AbortException` everywhere we handle `Exception` in Jenkins, I don't think that's reasonable idea.
Let's consider another reasonable way. e.g. overriding `AbortException#printStacktrace`, providing a exception-printing method to `BuildListener`.